lib/ramble/ramble/cmd/common/info.py

```diff
@@ -6,34 +6,306 @@
 # option. This file may not be copied, modified, or distributed
 # except according to those terms.
 
-from __future__ import print_function
+import fnmatch
 
 import llnl.util.tty.color as color
+from llnl.util.tty.colify import colified
 
+import ramble.util.colors
 import ramble.cmd.common.arguments as arguments
 import ramble.repository
 
-header_color = "@*b"
-plain_format = "@."
+from ramble.util.logger import logger
+
+import enum
+
+supported_formats = enum.Enum("formats", ["text", "lists"])
+
+obj_attribute_map = {
+    "maintainers": None,
+    "tags": None,
+    "pipelines": "_pipelines",
+    "figure_of_merit_contexts": None,
+    "figures_of_merit": None,
+    "builtins": None,
+    "package_manager_configs": None,
+    "required_packages": None,
+    "compilers": None,
+    "software_specs": None,
+    "archive_patterns": None,
+    "success_criteria": None,
+    # Application specific:
+    "workloads": None,
+    "workload_groups": None,
+    "executables": None,
+    "inputs": None,
+    "workload_variables": None,
+    "registered_phases": "phase_definitions",
+    # Modifier specific:
+    "modes": None,
+    "default_mode": "_default_usage_mode",
+    "variable_modifications": None,
+    "executable_modifiers": None,
+    "env_var_modifications": None,
+    "required_vars": None,
+    "modifier_variables": None,
+    "package_manager_requirements": None,
+    # Package manager specific:
+    "package_manager_variables": None,
+}
+
+
+def _map_attr_name(attr):
+    """Given an attribute display name, map to the internal attribute name"""
+    return obj_attribute_map[attr] if obj_attribute_map[attr] is not None else attr
 
 
 def setup_info_parser(subparser):
+    """Create the info parser"""
+
     subparser.add_argument("object", help="Name of object to print info for")
 
     arguments.add_common_arguments(subparser, ["obj_type"])
 
+    available_formats = []
+    for format in supported_formats:
+        available_formats.append(format.name)
+
+    subparser.add_argument(
+        "--format",
+        default="text",
+        help=f"list of output format to write. Available options are {available_formats}",
+    )
+
+    subparser.add_argument(
+        "--pattern",
+        "-p",
+        default="*",
+        metavar="PATTERN",
+        help="filter the attributes using this pattern",
+    )
+
+    level = subparser.add_mutually_exclusive_group()
+    level.add_argument(
+        "--overview",
+        "-o",
+        action="store_true",
+        help="print overview of what attributes an object has",
+    )
+    level.add_argument(
+        "--verbose", "-v", action="store_true", help="print full information about attributes"
+    )
+
+    attributes = subparser.add_mutually_exclusive_group()
+    attributes.add_argument(
+        "--all", action="store_true", default=True, help="print all object attributes"
+    )
+    attributes.add_argument(
+        "--attributes",
```

Comment:
Haha, being lazy, how about also having a shortcut like `--attrs`?

Reply:
Added both `--attrs` and `-a`. :)

Reply:
I thought about recommending `-a`, but then that seems to somewhat conflate with `--all`, and also that doesn't express the "plural" nature. WDYT?

Reply:
Ah sure, those are good points. I can pull `-a` back out.